homeassistant/components/yolink/sensor.py

```diff
@@ -10,31 +13,46 @@
     SensorStateClass,
 )
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_STATE, PERCENTAGE
-from homeassistant.core import HomeAssistant
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.util import percentage
 
+from . import YoLinkCoordinator
 from .const import (
     ATTR_CLIENT,
+    ATTR_COORDINATOR,
     ATTR_DEVICE,
     ATTR_DEVICE_DOOR_SENSOR,
+    ATTR_DEVICE_ID,
+    ATTR_DEVICE_NAME,
     ATTR_DEVICE_TYPE,
-    ATTR_MQTT_CLIENT,
-    ATTR_PLATFORM_SENSOR,
     DOMAIN,
 )
 from .entity import YoLinkEntity
 
-SENSOR_DESC_MAP: dict[str, SensorEntityDescription] = {
-    "battery": SensorEntityDescription(
+
+@dataclass
+class YoLinkSensorEntityDescription(SensorEntityDescription):
+    """YoLink SensorEntityDescription."""
+
+    value: Callable = round
+    supports: list[str] | None = None
```

Comment:
Why would we have a description that isn't supported? Make this attribute required, I think. Required attributes can be added to the sensor description by adding a separate mixin dataclass that we inherit in our sensor description class.

Example:
https://github.com/home-assistant/core/blob/4408ad82eb26f0b64b775d9ab60c00fd53820067/homeassistant/components/wled/sensor.py#L34-L47

Reply:
we have several different kind sensors, door sensor, temperature and humidity sensor, etc 
for door sensor, i want to export battery entity to sensor platform
for temperature and humidity sensor, i want to export battery, temperature, humidity entities to sensor platform

Reply:
Right. I'm saying that every entity description needs to have at least one device type that supports this kind of entity. Otherwise there's no point in adding the entity description.

Reply:
This still needs to be addressed.